src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -476,22 +475,68 @@ TypeSymbol BindRecursivePatternType(TypeSyntax typeSyntax, TypeSymbol inputType,
             }
         }
 
+        // Work around https://github.com/dotnet/roslyn/issues/20648: The compiler's internal APIs such as `declType.IsTupleType`
+        // do not correctly treat the non-generic struct `System.ValueTuple` as a tuple type.  We explicitly perform the tests
+        // required to identify it.  When that bug is fixed we should be able to remove this code and its callers.
+        internal static bool IsZeroElementTupleType(TypeSymbol type)
+        {
+            return type.IsStructType() && type.Name == "ValueTuple" && type.GetArity() == 0 && type.ContainingSymbol is var declContainer &&
+                declContainer.Kind == SymbolKind.Namespace && declContainer.Name == "System" && declContainer.ContainingSymbol?.Kind == SymbolKind.Namespace &&
+                declContainer.ContainingSymbol.ContainingSymbol?.Kind == SymbolKind.NetModule;
```

Comment:
Checking for `NetModule` seems unnecessary. Could we simply check `(declContainer.ContainingSymbol as NamespaceSymbol)?.IsGlobalNamespace == true`?